### PR TITLE
Fix x86 depthwise conv2d alter_op_layout

### PR DESCRIPTION
**Pull request by kevinthesun**
Fix x86 depthwise conv2d alter_op_layout. Related discussion topic: https://discuss.tvm.ai/t/x86-relay-auto-tune-mobilefacenet-error-using-relay/2508/22

### CODE REVIEW

File: topi/python/topi/x86/conv2d.py [hunk 1]
```diff
@@ -415,7 +415,8 @@ def _alter_conv2d_layout(attrs, inputs, tinfo, F):
 
     dtype = data.dtype
     out_dtype = dtype if out_dtype in ("same", "") else out_dtype
-    is_depthwise = groups == in_channel and groups == out_channel
+    kshape = get_const_tuple(kernel.shape)
+    is_depthwise = groups == kshape[0] and kshape[1] == 1
```
Comment (yzhliu) [inline topi/python/topi/x86/conv2d.py]: do we need to check kernel layout?

File: topi/python/topi/x86/conv2d.py [hunk 2]
```diff
@@ -415,11 +415,15 @@ def _alter_conv2d_layout(attrs, inputs, tinfo, F):
 
     dtype = data.dtype
     out_dtype = dtype if out_dtype in ("same", "") else out_dtype
-    is_depthwise = groups == in_channel and groups == out_channel
 
     # only optimize for NCHW
     if layout != 'NCHW':
```
Comment (yzhliu) [inline topi/python/topi/x86/conv2d.py]:
```suggestion
    if layout != 'NCHW' or attrs["kernel_layout"] != "OIHW":
```
and remove the assert below.

**Comment by yzhliu** [comment]
Thanks @kevinthesun !

**Comment by apivovarov** [comment]
@kevinthesun Looks like this PR makes autotvm to "skips depthwise_conv2d_nchw workloads for llvm x86_64"
More details: https://discuss.tvm.ai/t/autotvm-skips-depthwise-conv2d-nchw-workloads-for-llvm-x86-64/3353/1

**Comment by apivovarov** [comment]
https://github.com/dmlc/tvm/issues/3557

**Pull request merged**